src/backend/profile/profile.service.ts

```diff
@@ -100,3 +101,112 @@ export async function createUserProfile(profileData: CreateProfilePayload) {
   })
   return createdUser
 }
+
+async function updateUserCity(userId: string, newCity: City): Promise<void> {
+  if (!newCity) return
+
+  await prisma.profile.update({
+    where: {
+      userId,
+    },
+    data: {
+      city: {
+        connectOrCreate: {
+          where: {
+            id: newCity?.id,
+          },
+          create: {
+            name: newCity.name,
+            openForRelocation: true,
+          },
+        },
+      },
+    },
+  })
+}
+async function updateUserCountry(
+  userId: string,
+  newCountry: Country,
+): Promise<void> {
+  if (!newCountry) return
+
+  await prisma.profile.update({
+    where: {
+      userId,
+    },
+    data: {
+      country: {
+        connectOrCreate: {
+          where: {
+            id: newCountry?.id,
+          },
+          create: {
+            name: newCountry.name,
+            openForRelocation: true,
+          },
+        },
+      },
+    },
+  })
+}
+export async function updateUserData(
+  id: string,
+  userDataToUpdate: UpdateProfilePayload,
+) {
+  // take oldUserData for comparision
+  const user = await getProfileById(id)
+
+  // take city&country and update when is new
+  const oldCity = user?.city.name
+  const newCity = userDataToUpdate.city?.name
+  const oldCountry = user?.country.name
+  const newCountry = userDataToUpdate.country?.name
+
+  if (newCity !== oldCity) {
+    const foundCity = await prisma.city.findFirst({
```

Comment:
**nitpick:** This update should not be dependent on found city, because if such city does not exist, it should be created